src/core/lib/resolver/resolver.h

```diff
@@ -67,6 +69,14 @@ class Resolver : public InternallyRefCounted<Resolver> {
     // TODO(roth): Before making this a public API, figure out a way to
     // avoid exposing channel args this way.
     ChannelArgs args;
+    // If non-null, this callback will be invoked when the LB policy has
```

Comment:
Could we have cases where there is a race between the scheduling of the health callback and `ReportResult`? I'm thinking of a possibility where a callback gets delayed due to scheduling through something like an ExecCtx.

Reply:
I don't see how that could cause a race.  If the resolver chooses to return a callback, it cannot return another result until the callback is actually invoked, including whatever scheduling delay there is.  From the perspective of the resolver, it can't tell the difference between scheduling delay and the channel choosing to not invoke the callback yet for other reasons.

Reply:
Is there any additional synchronization requirement for the result_health_callback (things like locks/work serializer should be held or not held)?
Let's also document why `ReportResult` cannot return this status synchronously

Reply:
Updated comments.